pkg/cloudprovider/aws/instancetype.go

```diff
@@ -169,8 +169,13 @@ func (i *InstanceType) memory() *resource.Quantity {
 
 // Setting ephemeral-storage to be either the default value or what is defined in blockDeviceMappings
 func (i *InstanceType) ephemeralStorage() *resource.Quantity {
-	ephemeralBlockDevice := amifamily.GetAMIFamily(i.provider.AMIFamily, &amifamily.Options{}).EphemeralBlockDevice()
 	if i.provider.BlockDeviceMappings != nil {
```

Comment:
Checking length here seems a little safer although block device mappings shouldn't be an empty slice.  Line 176 would panic on an out-of-bounds error if it somehow was an empty slice though. 

```suggestion
	if len(i.provider.BlockDeviceMappings) != 0 {
```

Reply:
Make sense!